src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.go

```diff
@@ -1,43 +1,285 @@
 package storage_power_consensus
 
 import (
-	filcrypto "github.com/filecoin-project/specs/libraries/filcrypto"
+	libp2p "github.com/filecoin-project/specs/libraries/libp2p"
 	block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
-	deal "github.com/filecoin-project/specs/systems/filecoin_markets/deal"
-	sector "github.com/filecoin-project/specs/systems/filecoin_mining/sector"
+	actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
 	addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
 	util "github.com/filecoin-project/specs/util"
 )
 
-const PLEDGE_COLLATERAL_PER_GB = -1 // TODO define
+func (spa *StoragePowerActor_I) CreateStorageMiner(
+	ownerAddr addr.Address,
+	workerAddr addr.Address,
+	peerId libp2p.PeerID,
+) addr.Address {
+	var msgSender addr.Address // TODO replace this
 
-// Actor
-func (spa *StoragePowerActor_I) ReportConsensusFault(slasherAddr addr.Address, faultType ConsensusFaultType, proof []block.Block) {
-	panic("TODO")
+	// TODO: anything to check here?
+	newMiner := &PowerTableEntry_I{
+		ActivePower_:            block.StoragePower(0),
+		InactivePower_:          block.StoragePower(0),
+		AvailableBalance_:       actor.TokenAmount(0),
+		LockedPledgeCollateral_: actor.TokenAmount(0),
+	}
+	spa.PowerTable()[msgSender] = newMiner
```

Comment:
`msgSender` doesn't seem like the right key here, it should be the create actor's address.